src/Orleans/Core/GrainClient.cs

```diff
@@ -21,6 +21,13 @@ namespace Orleans
     public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
 
     /// <summary>
+    /// The delegate called before every request to a grain.
+    /// </summary>
+    /// <param name="request">The request.</param>
+    /// <param name="grain">The grain.</param>
+    public delegate void ClientInvokeCallback(InvokeMethodRequest request, IGrain grain);
```

Comment:
I moved the declaration from ClusterClient.cs to here, where it can live alongside the event handler definition, but I could also move it to its own file.